packaging/shark-platform/pyproject.toml

```diff
@@ -0,0 +1,42 @@
+[build-system]
+requires = ["setuptools", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "shark-platform"
```

Comment:
Shall we build this as part of https://github.com/nod-ai/SHARK-Platform/blob/main/.github/workflows/build_packages.yml too?

Reply:
I want releasing to be as automated as possible, so the developer / release engineer that is managing the process has to run a minimal number of commands and make as few decisions as possible.

* For the iree-org/iree packages, the process is to run https://github.com/iree-org/iree/blob/main/build_tools/python_deploy/pypi_deploy.sh pointed at a github release (one nightly release per day, so each release has only 1 version of each whl)
* For iree-org/iree-turbine, https://github.com/iree-org/iree-turbine/blob/main/docs/releasing.md has a sequence of commands to run: build package, test, upload to pypi. The build package step there involves a decision about the package version.
* For nod-ai/SHARK-Platform, we have scripts to build the wheels locally and dev wheels (from multiple versions) uploaded to https://github.com/nod-ai/SHARK-Platform/releases/tag/dev-wheels. We can either download from the dev-wheels or build locally, then update the package version, then publish.

(As an aside, I also want the dependency updating to be automated too - I lost most of Friday debugging dependency update issues in iree-turbine since we have no automation for that and things were broken. The release engineer should _just_ be updating packages and publishing, they shouldn't also have to debug the code)

Reply:
So... idk what specific recommendation I have here. We can start with a developer needing to build the meta package manually when publishing, but we should also be able to have test workflows in this project that pip install the nightly meta package and test it.

Reply:
Hmmm... could have a script that reads the release index for https://github.com/nod-ai/SHARK-Platform/releases/tag/dev-wheels and lets you pick a specific version to download, promote, and push. Could default to the latest nightly, prompt from a list, or require specifying explicitly (e.g. `2.9.0rc20241111` for today)

Reply:
I've now drafted it in a similar fashion to what I've written for `iree`. The script `build_tools/python_deploy/compute_common_version.py` computes a common version and writes it to a `version_local.json` file which is then parsed via a `setup.py`. Furthermore, the requirements file is also auto-generated as this feels more convenient. If we want to, we can simply add building the meta package to what is already existing in the CI.